exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Query.kt

```diff
@@ -373,6 +420,15 @@ open class Query(override var set: FieldSet, where: Op<Boolean>?) : AbstractQuer
             limit = oldLimit
         }
     }
+
+    /** Represents the position at which an SQL comment will be added in a `SELECT` query. */
+    enum class CommentPosition {
+        /** The start of the query, before the keyword `SELECT`. */
+        FRONT,
+
+        /** The end of the query, after all clauses. */
+        BACK
+    }
```

Comment:
I considered changing this to `START` and `END` but couldn't find many similar examples and figured it was still clear.

Reply:
@e5l  @bog-walk Could we add an option to insert a comment after the SELECT keyword to support [Optimizer Hints](https://dev.mysql.com/doc/refman/8.4/en/optimizer-hints.html)?

Reply:
Hi @fokkoru Yup, that's a popular and useful case, so I could add something like `CommentPosition.AFTER_SELECT`.
In the meantime though, if you need it, here's a custom query for inspiration:
```kt
fun Query.optimizerHint(hint: String) = OptimizerHintQuery(this, hint)

class OptimizerHintQuery(val source: Query, val optimizerHint: String) : Query(source.set, source.where) {
    init {
        source.copyTo(this)
    }

    override fun prepareSQL(builder: QueryBuilder): String {
        val originalSql = super.prepareSQL(builder)
        return originalSql.replace("SELECT ", "SELECT /*$optimizerHint*/ ")
    }

    override fun copy(): OptimizerHintQuery = OptimizerHintQuery(source.copy(), optimizerHint).also { copy ->
        copyTo(copy)
    }
}
```